api/document/js/keyboard.ts

```diff
@@ -1,16 +1,41 @@
-import { baseKeymap } from 'prosemirror-commands';
+import {
+  baseKeymap,
+  chainCommands,
+  deleteSelection,
+  selectNodeBackward,
+  selectNodeForward,
+} from 'prosemirror-commands';
 import { undo, redo } from 'prosemirror-history';
 import { keymap } from 'prosemirror-keymap';
 
 import Api from './Api';
 import { makeSave } from './commands';
 import schema from './schema';
 
+// Removing joinBackwards/forwards from actions taken when deleting as they
+// need to be tuned a bit better to our use case.
+const backspace = chainCommands(deleteSelection, selectNodeBackward);
+const del = chainCommands(deleteSelection, selectNodeForward);
+
 export default function menu(api: Api) {
   return keymap({
     ...baseKeymap,
+    // tslint:disable-next-line object-literal-key-quotes
+    'Backspace': backspace,
```

Comment:
FWIW, I quite dislike this particular `object-literal-key-quotes` tslint directive, because of how it acts in complete disregard of the context of the code it's placed in.  Particularly when we're force-disabling it through a very hard-to-remember comment directive like this, I'd rather just get rid of the rule than have to constantly override it (and honestly, even if the rule doesn't end up getting enforced in situations where it could/should be, I don't think it's that big a deal).

Reply:
Sure, we can turn it off.